// square/dist/esm/models/v1Payment.js
import { array, boolean, lazy, nullable, object, optional, string, } from '../schema';
import { v1DeviceSchema } from './v1Device';
import { v1MoneySchema } from './v1Money';
import { v1PaymentItemizationSchema, } from './v1PaymentItemization';
import { v1PaymentSurchargeSchema, } from './v1PaymentSurcharge';
import { v1PaymentTaxSchema } from './v1PaymentTax';
import { v1RefundSchema } from './v1Refund';
import { v1TenderSchema } from './v1Tender';
export const v1PaymentSchema = object({
    id: ['id', optional(string())],
    merchantId: ['merchant_id', optional(nullable(string()))],
    createdAt: ['created_at', optional(string())],
    creatorId: ['creator_id', optional(nullable(string()))],
    device: ['device', optional(lazy(() => v1DeviceSchema))],
    paymentUrl: ['payment_url', optional(nullable(string()))],
    receiptUrl: ['receipt_url', optional(nullable(string()))],
    inclusiveTaxMoney: [
        'inclusive_tax_money',
        optional(lazy(() => v1MoneySchema)),
    ],
    additiveTaxMoney: ['additive_tax_money', optional(lazy(() => v1MoneySchema))],
    taxMoney: ['tax_money', optional(lazy(() => v1MoneySchema))],
    tipMoney: ['tip_money', optional(lazy(() => v1MoneySchema))],
    discountMoney: ['discount_money', optional(lazy(() => v1MoneySchema))],
    totalCollectedMoney: [
        'total_collected_money',
        optional(lazy(() => v1MoneySchema)),
    ],
    processingFeeMoney: [
        'processing_fee_money',
        optional(lazy(() => v1MoneySchema)),
    ],
    netTotalMoney: ['net_total_money', optional(lazy(() => v1MoneySchema))],
    refundedMoney: ['refunded_money', optional(lazy(() => v1MoneySchema))],
    swedishRoundingMoney: [
        'swedish_rounding_money',
        optional(lazy(() => v1MoneySchema)),
    ],
    grossSalesMoney: ['gross_sales_money', optional(lazy(() => v1MoneySchema))],
    netSalesMoney: ['net_sales_money', optional(lazy(() => v1MoneySchema))],
    inclusiveTax: [
        'inclusive_tax',
        optional(nullable(array(lazy(() => v1PaymentTaxSchema)))),
    ],
    additiveTax: [
        'additive_tax',
        optional(nullable(array(lazy(() => v1PaymentTaxSchema)))),
    ],
    tender: ['tender', optional(nullable(array(lazy(() => v1TenderSchema))))],
    refunds: ['refunds', optional(nullable(array(lazy(() => v1RefundSchema))))],
    itemizations: [
        'itemizations',
        optional(nullable(array(lazy(() => v1PaymentItemizationSchema)))),
    ],
    surchargeMoney: ['surcharge_money', optional(lazy(() => v1MoneySchema))],
    surcharges: [
        'surcharges',
        optional(nullable(array(lazy(() => v1PaymentSurchargeSchema)))),
    ],
    isPartial: ['is_partial', optional(nullable(boolean()))],
});
