// @ionos-cloud/sdk-nodejs/dist/api/flow-logs-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowLogsApi = exports.FlowLogsApiFactory = exports.FlowLogsApiFp = exports.FlowLogsApiAxiosParamCreator = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
var base_1 = require("../base");
/**
 * FlowLogsApi - axios parameter creator
 * @export
 */
exports.FlowLogsApiAxiosParamCreator = function (configuration) {
    var _this = this;
    return {
        /**
         * Delete the specified Flow Log.
         * @summary Delete Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsDelete: function (datacenterId, serverId, nicId, flowlogId, pretty, depth, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersServersNicsFlowlogsDelete.');
                            }
                            if (serverId === null || serverId === undefined) {
                                throw new base_1.RequiredError('serverId', 'Required parameter serverId was null or undefined when calling datacentersServersNicsFlowlogsDelete.');
                            }
                            if (nicId === null || nicId === undefined) {
                                throw new base_1.RequiredError('nicId', 'Required parameter nicId was null or undefined when calling datacentersServersNicsFlowlogsDelete.');
                            }
                            if (flowlogId === null || flowlogId === undefined) {
                                throw new base_1.RequiredError('flowlogId', 'Required parameter flowlogId was null or undefined when calling datacentersServersNicsFlowlogsDelete.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/servers/{serverId}/nics/{nicId}/flowlogs/{flowlogId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "serverId" + "}", encodeURIComponent(String(serverId)))
                                .replace("{" + "nicId" + "}", encodeURIComponent(String(nicId)))
                                .replace("{" + "flowlogId" + "}", encodeURIComponent(String(flowlogId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'DELETE' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieve the properties of the specified Flow Log.
         * @summary Retrieve Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsFindById: function (datacenterId, serverId, nicId, flowlogId, pretty, depth, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersServersNicsFlowlogsFindById.');
                            }
                            if (serverId === null || serverId === undefined) {
                                throw new base_1.RequiredError('serverId', 'Required parameter serverId was null or undefined when calling datacentersServersNicsFlowlogsFindById.');
                            }
                            if (nicId === null || nicId === undefined) {
                                throw new base_1.RequiredError('nicId', 'Required parameter nicId was null or undefined when calling datacentersServersNicsFlowlogsFindById.');
                            }
                            if (flowlogId === null || flowlogId === undefined) {
                                throw new base_1.RequiredError('flowlogId', 'Required parameter flowlogId was null or undefined when calling datacentersServersNicsFlowlogsFindById.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/servers/{serverId}/nics/{nicId}/flowlogs/{flowlogId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "serverId" + "}", encodeURIComponent(String(serverId)))
                                .replace("{" + "nicId" + "}", encodeURIComponent(String(nicId)))
                                .replace("{" + "flowlogId" + "}", encodeURIComponent(String(flowlogId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * List all the Flow Logs for the specified NIC.
         * @summary List Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsGet: function (datacenterId, serverId, nicId, pretty, depth, offset, limit, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersServersNicsFlowlogsGet.');
                            }
                            if (serverId === null || serverId === undefined) {
                                throw new base_1.RequiredError('serverId', 'Required parameter serverId was null or undefined when calling datacentersServersNicsFlowlogsGet.');
                            }
                            if (nicId === null || nicId === undefined) {
                                throw new base_1.RequiredError('nicId', 'Required parameter nicId was null or undefined when calling datacentersServersNicsFlowlogsGet.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/servers/{serverId}/nics/{nicId}/flowlogs"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "serverId" + "}", encodeURIComponent(String(serverId)))
                                .replace("{" + "nicId" + "}", encodeURIComponent(String(nicId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((offset === undefined) && (configuration !== undefined)) {
                                offset = configuration.getDefaultParamValue('offset');
                            }
                            if (offset !== undefined) {
                                localVarQueryParameter['offset'] = offset;
                            }
                            if ((limit === undefined) && (configuration !== undefined)) {
                                limit = configuration.getDefaultParamValue('limit');
                            }
                            if (limit !== undefined) {
                                localVarQueryParameter['limit'] = limit;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Update the specified Flow Log record.
         * @summary Partially modify Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {FlowLogProperties} flowlog The Flow Log record to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPatch: function (datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersServersNicsFlowlogsPatch.');
                            }
                            if (serverId === null || serverId === undefined) {
                                throw new base_1.RequiredError('serverId', 'Required parameter serverId was null or undefined when calling datacentersServersNicsFlowlogsPatch.');
                            }
                            if (nicId === null || nicId === undefined) {
                                throw new base_1.RequiredError('nicId', 'Required parameter nicId was null or undefined when calling datacentersServersNicsFlowlogsPatch.');
                            }
                            if (flowlogId === null || flowlogId === undefined) {
                                throw new base_1.RequiredError('flowlogId', 'Required parameter flowlogId was null or undefined when calling datacentersServersNicsFlowlogsPatch.');
                            }
                            if (flowlog === null || flowlog === undefined) {
                                throw new base_1.RequiredError('flowlog', 'Required parameter flowlog was null or undefined when calling datacentersServersNicsFlowlogsPatch.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/servers/{serverId}/nics/{nicId}/flowlogs/{flowlogId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "serverId" + "}", encodeURIComponent(String(serverId)))
                                .replace("{" + "nicId" + "}", encodeURIComponent(String(nicId)))
                                .replace("{" + "flowlogId" + "}", encodeURIComponent(String(flowlogId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'PATCH' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof flowlog !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(flowlog !== undefined ? flowlog : {})
                                : (flowlog || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Adds a new Flow Log for the specified NIC.
         * @summary Create a Flow Log
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {FlowLog} flowlog The Flow Log to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPost: function (datacenterId, serverId, nicId, flowlog, pretty, depth, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersServersNicsFlowlogsPost.');
                            }
                            if (serverId === null || serverId === undefined) {
                                throw new base_1.RequiredError('serverId', 'Required parameter serverId was null or undefined when calling datacentersServersNicsFlowlogsPost.');
                            }
                            if (nicId === null || nicId === undefined) {
                                throw new base_1.RequiredError('nicId', 'Required parameter nicId was null or undefined when calling datacentersServersNicsFlowlogsPost.');
                            }
                            if (flowlog === null || flowlog === undefined) {
                                throw new base_1.RequiredError('flowlog', 'Required parameter flowlog was null or undefined when calling datacentersServersNicsFlowlogsPost.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/servers/{serverId}/nics/{nicId}/flowlogs"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "serverId" + "}", encodeURIComponent(String(serverId)))
                                .replace("{" + "nicId" + "}", encodeURIComponent(String(nicId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof flowlog !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(flowlog !== undefined ? flowlog : {})
                                : (flowlog || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Modify the specified Flow Log record.
         * @summary Modify Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {FlowLogPut} flowlog The modified Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPut: function (datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersServersNicsFlowlogsPut.');
                            }
                            if (serverId === null || serverId === undefined) {
                                throw new base_1.RequiredError('serverId', 'Required parameter serverId was null or undefined when calling datacentersServersNicsFlowlogsPut.');
                            }
                            if (nicId === null || nicId === undefined) {
                                throw new base_1.RequiredError('nicId', 'Required parameter nicId was null or undefined when calling datacentersServersNicsFlowlogsPut.');
                            }
                            if (flowlogId === null || flowlogId === undefined) {
                                throw new base_1.RequiredError('flowlogId', 'Required parameter flowlogId was null or undefined when calling datacentersServersNicsFlowlogsPut.');
                            }
                            if (flowlog === null || flowlog === undefined) {
                                throw new base_1.RequiredError('flowlog', 'Required parameter flowlog was null or undefined when calling datacentersServersNicsFlowlogsPut.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/servers/{serverId}/nics/{nicId}/flowlogs/{flowlogId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "serverId" + "}", encodeURIComponent(String(serverId)))
                                .replace("{" + "nicId" + "}", encodeURIComponent(String(nicId)))
                                .replace("{" + "flowlogId" + "}", encodeURIComponent(String(flowlogId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'PUT' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof flowlog !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(flowlog !== undefined ? flowlog : {})
                                : (flowlog || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
    };
};
/**
 * FlowLogsApi - functional programming interface
 * @export
 */
exports.FlowLogsApiFp = function (configuration) {
    return {
        /**
         * Delete the specified Flow Log.
         * @summary Delete Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsDelete: function (datacenterId, serverId, nicId, flowlogId, pretty, depth, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.FlowLogsApiAxiosParamCreator(configuration).datacentersServersNicsFlowlogsDelete(datacenterId, serverId, nicId, flowlogId, pretty, depth, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieve the properties of the specified Flow Log.
         * @summary Retrieve Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsFindById: function (datacenterId, serverId, nicId, flowlogId, pretty, depth, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.FlowLogsApiAxiosParamCreator(configuration).datacentersServersNicsFlowlogsFindById(datacenterId, serverId, nicId, flowlogId, pretty, depth, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * List all the Flow Logs for the specified NIC.
         * @summary List Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsGet: function (datacenterId, serverId, nicId, pretty, depth, offset, limit, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.FlowLogsApiAxiosParamCreator(configuration).datacentersServersNicsFlowlogsGet(datacenterId, serverId, nicId, pretty, depth, offset, limit, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Update the specified Flow Log record.
         * @summary Partially modify Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {FlowLogProperties} flowlog The Flow Log record to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPatch: function (datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.FlowLogsApiAxiosParamCreator(configuration).datacentersServersNicsFlowlogsPatch(datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Adds a new Flow Log for the specified NIC.
         * @summary Create a Flow Log
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {FlowLog} flowlog The Flow Log to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPost: function (datacenterId, serverId, nicId, flowlog, pretty, depth, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.FlowLogsApiAxiosParamCreator(configuration).datacentersServersNicsFlowlogsPost(datacenterId, serverId, nicId, flowlog, pretty, depth, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Modify the specified Flow Log record.
         * @summary Modify Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {FlowLogPut} flowlog The modified Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPut: function (datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.FlowLogsApiAxiosParamCreator(configuration).datacentersServersNicsFlowlogsPut(datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
    };
};
/**
 * FlowLogsApi - factory interface
 * @export
 */
exports.FlowLogsApiFactory = function (configuration, basePath, axios) {
    return {
        /**
         * Delete the specified Flow Log.
         * @summary Delete Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsDelete: function (datacenterId, serverId, nicId, flowlogId, pretty, depth, options) {
            return exports.FlowLogsApiFp(configuration).datacentersServersNicsFlowlogsDelete(datacenterId, serverId, nicId, flowlogId, pretty, depth, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieve the properties of the specified Flow Log.
         * @summary Retrieve Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsFindById: function (datacenterId, serverId, nicId, flowlogId, pretty, depth, options) {
            return exports.FlowLogsApiFp(configuration).datacentersServersNicsFlowlogsFindById(datacenterId, serverId, nicId, flowlogId, pretty, depth, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * List all the Flow Logs for the specified NIC.
         * @summary List Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsGet: function (datacenterId, serverId, nicId, pretty, depth, offset, limit, options) {
            return exports.FlowLogsApiFp(configuration).datacentersServersNicsFlowlogsGet(datacenterId, serverId, nicId, pretty, depth, offset, limit, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Update the specified Flow Log record.
         * @summary Partially modify Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {FlowLogProperties} flowlog The Flow Log record to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPatch: function (datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options) {
            return exports.FlowLogsApiFp(configuration).datacentersServersNicsFlowlogsPatch(datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Adds a new Flow Log for the specified NIC.
         * @summary Create a Flow Log
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {FlowLog} flowlog The Flow Log to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPost: function (datacenterId, serverId, nicId, flowlog, pretty, depth, options) {
            return exports.FlowLogsApiFp(configuration).datacentersServersNicsFlowlogsPost(datacenterId, serverId, nicId, flowlog, pretty, depth, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Modify the specified Flow Log record.
         * @summary Modify Flow Logs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} serverId The unique ID of the server.
         * @param {string} nicId The unique ID of the NIC.
         * @param {string} flowlogId The unique ID of the Flow Log.
         * @param {FlowLogPut} flowlog The modified Flow Log.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersServersNicsFlowlogsPut: function (datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options) {
            return exports.FlowLogsApiFp(configuration).datacentersServersNicsFlowlogsPut(datacenterId, serverId, nicId, flowlogId, flowlog, pretty, depth, options).then(function (request) { return request(axios, basePath); });
        },
    };
};
/**
 * FlowLogsApi - object-oriented interface
 * @export
 * @class FlowLogsApi
 * @extends {BaseAPI}
 */
var FlowLogsApi = /** @class */ (function (_super) {
    __extends(FlowLogsApi, _super);
    function FlowLogsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Delete the specified Flow Log.
     * @summary Delete Flow Logs
     * @param {FlowLogsApiDatacentersServersNicsFlowlogsDeleteRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowLogsApi
     */
    FlowLogsApi.prototype.datacentersServersNicsFlowlogsDelete = function (requestParameters, options) {
        var _this = this;
        return exports.FlowLogsApiFp(this.configuration).datacentersServersNicsFlowlogsDelete(requestParameters.datacenterId, requestParameters.serverId, requestParameters.nicId, requestParameters.flowlogId, requestParameters.pretty, requestParameters.depth, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieve the properties of the specified Flow Log.
     * @summary Retrieve Flow Logs
     * @param {FlowLogsApiDatacentersServersNicsFlowlogsFindByIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowLogsApi
     */
    FlowLogsApi.prototype.datacentersServersNicsFlowlogsFindById = function (requestParameters, options) {
        var _this = this;
        return exports.FlowLogsApiFp(this.configuration).datacentersServersNicsFlowlogsFindById(requestParameters.datacenterId, requestParameters.serverId, requestParameters.nicId, requestParameters.flowlogId, requestParameters.pretty, requestParameters.depth, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * List all the Flow Logs for the specified NIC.
     * @summary List Flow Logs
     * @param {FlowLogsApiDatacentersServersNicsFlowlogsGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowLogsApi
     */
    FlowLogsApi.prototype.datacentersServersNicsFlowlogsGet = function (requestParameters, options) {
        var _this = this;
        return exports.FlowLogsApiFp(this.configuration).datacentersServersNicsFlowlogsGet(requestParameters.datacenterId, requestParameters.serverId, requestParameters.nicId, requestParameters.pretty, requestParameters.depth, requestParameters.offset, requestParameters.limit, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Update the specified Flow Log record.
     * @summary Partially modify Flow Logs
     * @param {FlowLogsApiDatacentersServersNicsFlowlogsPatchRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowLogsApi
     */
    FlowLogsApi.prototype.datacentersServersNicsFlowlogsPatch = function (requestParameters, options) {
        var _this = this;
        return exports.FlowLogsApiFp(this.configuration).datacentersServersNicsFlowlogsPatch(requestParameters.datacenterId, requestParameters.serverId, requestParameters.nicId, requestParameters.flowlogId, requestParameters.flowlog, requestParameters.pretty, requestParameters.depth, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Adds a new Flow Log for the specified NIC.
     * @summary Create a Flow Log
     * @param {FlowLogsApiDatacentersServersNicsFlowlogsPostRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowLogsApi
     */
    FlowLogsApi.prototype.datacentersServersNicsFlowlogsPost = function (requestParameters, options) {
        var _this = this;
        return exports.FlowLogsApiFp(this.configuration).datacentersServersNicsFlowlogsPost(requestParameters.datacenterId, requestParameters.serverId, requestParameters.nicId, requestParameters.flowlog, requestParameters.pretty, requestParameters.depth, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Modify the specified Flow Log record.
     * @summary Modify Flow Logs
     * @param {FlowLogsApiDatacentersServersNicsFlowlogsPutRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowLogsApi
     */
    FlowLogsApi.prototype.datacentersServersNicsFlowlogsPut = function (requestParameters, options) {
        var _this = this;
        return exports.FlowLogsApiFp(this.configuration).datacentersServersNicsFlowlogsPut(requestParameters.datacenterId, requestParameters.serverId, requestParameters.nicId, requestParameters.flowlogId, requestParameters.flowlog, requestParameters.pretty, requestParameters.depth, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    return FlowLogsApi;
}(base_1.BaseAPI));
exports.FlowLogsApi = FlowLogsApi;
