// @ionos-cloud/sdk-nodejs/dist/api/kubernetes-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KubernetesApi = exports.KubernetesApiFactory = exports.KubernetesApiFp = exports.KubernetesApiAxiosParamCreator = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
var base_1 = require("../base");
/**
 * KubernetesApi - axios parameter creator
 * @export
 */
exports.KubernetesApiAxiosParamCreator = function (configuration) {
    var _this = this;
    return {
        /**
         * Deletes the K8s cluster specified  by its ID.
         * @summary Delete a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sDelete: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sDelete.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'DELETE' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves the K8s cluster specified by its ID.
         * @summary Get a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the K8s cluster to be retrieved.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sFindByClusterId: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sFindByClusterId.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves a list of all K8s clusters provisioned under your account.
         * @summary Get Kubernetes Clusters
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sGet: function (pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            localVarPath = "/k8s";
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves the configuration file for the specified K8s cluster. You can define the format (YAML or JSON) of the returned file in the Accept header. By default, \'application/yaml\' is specified.
         * @summary Get Kubernetes Configuration File
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sKubeconfigGet: function (k8sClusterId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sKubeconfigGet.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/kubeconfig"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Deletes the K8s node pool specified by its ID.
         * @summary Delete a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsDelete: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsDelete.');
                            }
                            if (nodepoolId === null || nodepoolId === undefined) {
                                throw new base_1.RequiredError('nodepoolId', 'Required parameter nodepoolId was null or undefined when calling k8sNodepoolsDelete.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools/{nodepoolId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)))
                                .replace("{" + "nodepoolId" + "}", encodeURIComponent(String(nodepoolId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'DELETE' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves the K8s node pool specified by its ID.
         * @summary Get a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsFindById: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsFindById.');
                            }
                            if (nodepoolId === null || nodepoolId === undefined) {
                                throw new base_1.RequiredError('nodepoolId', 'Required parameter nodepoolId was null or undefined when calling k8sNodepoolsFindById.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools/{nodepoolId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)))
                                .replace("{" + "nodepoolId" + "}", encodeURIComponent(String(nodepoolId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves a list of K8s node pools of a cluster specified by its ID.
         * @summary Get Kubernetes Node Pools
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsGet: function (k8sClusterId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsGet.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Deletes the K8s node specified by its ID.
         * @summary Delete a Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesDelete: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsNodesDelete.');
                            }
                            if (nodepoolId === null || nodepoolId === undefined) {
                                throw new base_1.RequiredError('nodepoolId', 'Required parameter nodepoolId was null or undefined when calling k8sNodepoolsNodesDelete.');
                            }
                            if (nodeId === null || nodeId === undefined) {
                                throw new base_1.RequiredError('nodeId', 'Required parameter nodeId was null or undefined when calling k8sNodepoolsNodesDelete.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools/{nodepoolId}/nodes/{nodeId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)))
                                .replace("{" + "nodepoolId" + "}", encodeURIComponent(String(nodepoolId)))
                                .replace("{" + "nodeId" + "}", encodeURIComponent(String(nodeId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'DELETE' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves the K8s node specified by its ID.
         * @summary Get Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesFindById: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsNodesFindById.');
                            }
                            if (nodepoolId === null || nodepoolId === undefined) {
                                throw new base_1.RequiredError('nodepoolId', 'Required parameter nodepoolId was null or undefined when calling k8sNodepoolsNodesFindById.');
                            }
                            if (nodeId === null || nodeId === undefined) {
                                throw new base_1.RequiredError('nodeId', 'Required parameter nodeId was null or undefined when calling k8sNodepoolsNodesFindById.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools/{nodepoolId}/nodes/{nodeId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)))
                                .replace("{" + "nodepoolId" + "}", encodeURIComponent(String(nodepoolId)))
                                .replace("{" + "nodeId" + "}", encodeURIComponent(String(nodeId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves the list of all K8s nodes of the specified node pool.
         * @summary Get Kubernetes Nodes
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesGet: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsNodesGet.');
                            }
                            if (nodepoolId === null || nodepoolId === undefined) {
                                throw new base_1.RequiredError('nodepoolId', 'Required parameter nodepoolId was null or undefined when calling k8sNodepoolsNodesGet.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools/{nodepoolId}/nodes"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)))
                                .replace("{" + "nodepoolId" + "}", encodeURIComponent(String(nodepoolId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Recreates the K8s node specified by its ID.  If a node becomes unusable, Managed Kubernetes allows you to recreate it with a configuration based on the node pool template. Once the status is \'Active,\' all the pods from the failed node will be migrated to the new node. The node pool has an additional billable \'active\' node during this process.
         * @summary Recreate a Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesReplacePost: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsNodesReplacePost.');
                            }
                            if (nodepoolId === null || nodepoolId === undefined) {
                                throw new base_1.RequiredError('nodepoolId', 'Required parameter nodepoolId was null or undefined when calling k8sNodepoolsNodesReplacePost.');
                            }
                            if (nodeId === null || nodeId === undefined) {
                                throw new base_1.RequiredError('nodeId', 'Required parameter nodeId was null or undefined when calling k8sNodepoolsNodesReplacePost.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools/{nodepoolId}/nodes/{nodeId}/replace"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)))
                                .replace("{" + "nodepoolId" + "}", encodeURIComponent(String(nodepoolId)))
                                .replace("{" + "nodeId" + "}", encodeURIComponent(String(nodeId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Creates a node pool inside the specified K8s cluster.
         * @summary Create a Kubernetes Node Pool
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {KubernetesNodePoolForPost} kubernetesNodePool The Kubernetes node pool to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsPost: function (k8sClusterId, kubernetesNodePool, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsPost.');
                            }
                            if (kubernetesNodePool === null || kubernetesNodePool === undefined) {
                                throw new base_1.RequiredError('kubernetesNodePool', 'Required parameter kubernetesNodePool was null or undefined when calling k8sNodepoolsPost.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof kubernetesNodePool !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(kubernetesNodePool !== undefined ? kubernetesNodePool : {})
                                : (kubernetesNodePool || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Modifies the K8s node pool specified by its ID.
         * @summary Modify a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {KubernetesNodePoolForPut} kubernetesNodePool Details of the Kubernetes Node Pool
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsPut: function (k8sClusterId, nodepoolId, kubernetesNodePool, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sNodepoolsPut.');
                            }
                            if (nodepoolId === null || nodepoolId === undefined) {
                                throw new base_1.RequiredError('nodepoolId', 'Required parameter nodepoolId was null or undefined when calling k8sNodepoolsPut.');
                            }
                            if (kubernetesNodePool === null || kubernetesNodePool === undefined) {
                                throw new base_1.RequiredError('kubernetesNodePool', 'Required parameter kubernetesNodePool was null or undefined when calling k8sNodepoolsPut.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}/nodepools/{nodepoolId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)))
                                .replace("{" + "nodepoolId" + "}", encodeURIComponent(String(nodepoolId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'PUT' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof kubernetesNodePool !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(kubernetesNodePool !== undefined ? kubernetesNodePool : {})
                                : (kubernetesNodePool || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Creates a K8s cluster provisioned under your account.
         * @summary Create a Kubernetes Cluster
         * @param {KubernetesClusterForPost} kubernetesCluster The Kubernetes cluster to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sPost: function (kubernetesCluster, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (kubernetesCluster === null || kubernetesCluster === undefined) {
                                throw new base_1.RequiredError('kubernetesCluster', 'Required parameter kubernetesCluster was null or undefined when calling k8sPost.');
                            }
                            localVarPath = "/k8s";
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof kubernetesCluster !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(kubernetesCluster !== undefined ? kubernetesCluster : {})
                                : (kubernetesCluster || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Modifies the K8s cluster specified by its ID.
         * @summary Modify a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {KubernetesClusterForPut} kubernetesCluster The modified Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sPut: function (k8sClusterId, kubernetesCluster, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (k8sClusterId === null || k8sClusterId === undefined) {
                                throw new base_1.RequiredError('k8sClusterId', 'Required parameter k8sClusterId was null or undefined when calling k8sPut.');
                            }
                            if (kubernetesCluster === null || kubernetesCluster === undefined) {
                                throw new base_1.RequiredError('kubernetesCluster', 'Required parameter kubernetesCluster was null or undefined when calling k8sPut.');
                            }
                            localVarPath = "/k8s/{k8sClusterId}"
                                .replace("{" + "k8sClusterId" + "}", encodeURIComponent(String(k8sClusterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'PUT' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof kubernetesCluster !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(kubernetesCluster !== undefined ? kubernetesCluster : {})
                                : (kubernetesCluster || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieves the current default K8s version to be used by the clusters and node pools.
         * @summary Get Default Kubernetes Version
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sVersionsDefaultGet: function (orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            localVarPath = "/k8s/versions/default";
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Lists available K8s versions.
         * @summary Get Kubernetes Versions
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sVersionsGet: function (orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            localVarPath = "/k8s/versions";
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
    };
};
/**
 * KubernetesApi - functional programming interface
 * @export
 */
exports.KubernetesApiFp = function (configuration) {
    return {
        /**
         * Deletes the K8s cluster specified  by its ID.
         * @summary Delete a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sDelete: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sDelete(k8sClusterId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves the K8s cluster specified by its ID.
         * @summary Get a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the K8s cluster to be retrieved.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sFindByClusterId: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sFindByClusterId(k8sClusterId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves a list of all K8s clusters provisioned under your account.
         * @summary Get Kubernetes Clusters
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sGet: function (pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sGet(pretty, depth, xContractNumber, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves the configuration file for the specified K8s cluster. You can define the format (YAML or JSON) of the returned file in the Accept header. By default, \'application/yaml\' is specified.
         * @summary Get Kubernetes Configuration File
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sKubeconfigGet: function (k8sClusterId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sKubeconfigGet(k8sClusterId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Deletes the K8s node pool specified by its ID.
         * @summary Delete a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsDelete: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsDelete(k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves the K8s node pool specified by its ID.
         * @summary Get a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsFindById: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsFindById(k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves a list of K8s node pools of a cluster specified by its ID.
         * @summary Get Kubernetes Node Pools
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsGet: function (k8sClusterId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsGet(k8sClusterId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Deletes the K8s node specified by its ID.
         * @summary Delete a Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesDelete: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsNodesDelete(k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves the K8s node specified by its ID.
         * @summary Get Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesFindById: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsNodesFindById(k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves the list of all K8s nodes of the specified node pool.
         * @summary Get Kubernetes Nodes
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesGet: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsNodesGet(k8sClusterId, nodepoolId, pretty, depth, xContractNumber, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Recreates the K8s node specified by its ID.  If a node becomes unusable, Managed Kubernetes allows you to recreate it with a configuration based on the node pool template. Once the status is \'Active,\' all the pods from the failed node will be migrated to the new node. The node pool has an additional billable \'active\' node during this process.
         * @summary Recreate a Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesReplacePost: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsNodesReplacePost(k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Creates a node pool inside the specified K8s cluster.
         * @summary Create a Kubernetes Node Pool
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {KubernetesNodePoolForPost} kubernetesNodePool The Kubernetes node pool to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsPost: function (k8sClusterId, kubernetesNodePool, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsPost(k8sClusterId, kubernetesNodePool, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Modifies the K8s node pool specified by its ID.
         * @summary Modify a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {KubernetesNodePoolForPut} kubernetesNodePool Details of the Kubernetes Node Pool
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsPut: function (k8sClusterId, nodepoolId, kubernetesNodePool, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sNodepoolsPut(k8sClusterId, nodepoolId, kubernetesNodePool, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Creates a K8s cluster provisioned under your account.
         * @summary Create a Kubernetes Cluster
         * @param {KubernetesClusterForPost} kubernetesCluster The Kubernetes cluster to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sPost: function (kubernetesCluster, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sPost(kubernetesCluster, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Modifies the K8s cluster specified by its ID.
         * @summary Modify a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {KubernetesClusterForPut} kubernetesCluster The modified Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sPut: function (k8sClusterId, kubernetesCluster, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sPut(k8sClusterId, kubernetesCluster, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieves the current default K8s version to be used by the clusters and node pools.
         * @summary Get Default Kubernetes Version
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sVersionsDefaultGet: function (orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sVersionsDefaultGet(orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Lists available K8s versions.
         * @summary Get Kubernetes Versions
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sVersionsGet: function (orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.KubernetesApiAxiosParamCreator(configuration).k8sVersionsGet(orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
    };
};
/**
 * KubernetesApi - factory interface
 * @export
 */
exports.KubernetesApiFactory = function (configuration, basePath, axios) {
    return {
        /**
         * Deletes the K8s cluster specified  by its ID.
         * @summary Delete a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sDelete: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sDelete(k8sClusterId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves the K8s cluster specified by its ID.
         * @summary Get a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the K8s cluster to be retrieved.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sFindByClusterId: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sFindByClusterId(k8sClusterId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves a list of all K8s clusters provisioned under your account.
         * @summary Get Kubernetes Clusters
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sGet: function (pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sGet(pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves the configuration file for the specified K8s cluster. You can define the format (YAML or JSON) of the returned file in the Accept header. By default, \'application/yaml\' is specified.
         * @summary Get Kubernetes Configuration File
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sKubeconfigGet: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sKubeconfigGet(k8sClusterId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Deletes the K8s node pool specified by its ID.
         * @summary Delete a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsDelete: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsDelete(k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves the K8s node pool specified by its ID.
         * @summary Get a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsFindById: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsFindById(k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves a list of K8s node pools of a cluster specified by its ID.
         * @summary Get Kubernetes Node Pools
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsGet: function (k8sClusterId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsGet(k8sClusterId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Deletes the K8s node specified by its ID.
         * @summary Delete a Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesDelete: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsNodesDelete(k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves the K8s node specified by its ID.
         * @summary Get Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesFindById: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsNodesFindById(k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves the list of all K8s nodes of the specified node pool.
         * @summary Get Kubernetes Nodes
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesGet: function (k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsNodesGet(k8sClusterId, nodepoolId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Recreates the K8s node specified by its ID.  If a node becomes unusable, Managed Kubernetes allows you to recreate it with a configuration based on the node pool template. Once the status is \'Active,\' all the pods from the failed node will be migrated to the new node. The node pool has an additional billable \'active\' node during this process.
         * @summary Recreate a Kubernetes Node by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {string} nodeId The unique ID of the Kubernetes node.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsNodesReplacePost: function (k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsNodesReplacePost(k8sClusterId, nodepoolId, nodeId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Creates a node pool inside the specified K8s cluster.
         * @summary Create a Kubernetes Node Pool
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {KubernetesNodePoolForPost} kubernetesNodePool The Kubernetes node pool to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsPost: function (k8sClusterId, kubernetesNodePool, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsPost(k8sClusterId, kubernetesNodePool, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Modifies the K8s node pool specified by its ID.
         * @summary Modify a Kubernetes Node Pool by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {string} nodepoolId The unique ID of the Kubernetes node pool.
         * @param {KubernetesNodePoolForPut} kubernetesNodePool Details of the Kubernetes Node Pool
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sNodepoolsPut: function (k8sClusterId, nodepoolId, kubernetesNodePool, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sNodepoolsPut(k8sClusterId, nodepoolId, kubernetesNodePool, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Creates a K8s cluster provisioned under your account.
         * @summary Create a Kubernetes Cluster
         * @param {KubernetesClusterForPost} kubernetesCluster The Kubernetes cluster to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sPost: function (kubernetesCluster, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sPost(kubernetesCluster, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Modifies the K8s cluster specified by its ID.
         * @summary Modify a Kubernetes Cluster by ID
         * @param {string} k8sClusterId The unique ID of the Kubernetes cluster.
         * @param {KubernetesClusterForPut} kubernetesCluster The modified Kubernetes cluster.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sPut: function (k8sClusterId, kubernetesCluster, pretty, depth, xContractNumber, options) {
            return exports.KubernetesApiFp(configuration).k8sPut(k8sClusterId, kubernetesCluster, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieves the current default K8s version to be used by the clusters and node pools.
         * @summary Get Default Kubernetes Version
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sVersionsDefaultGet: function (options) {
            return exports.KubernetesApiFp(configuration).k8sVersionsDefaultGet(options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Lists available K8s versions.
         * @summary Get Kubernetes Versions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        k8sVersionsGet: function (options) {
            return exports.KubernetesApiFp(configuration).k8sVersionsGet(options).then(function (request) { return request(axios, basePath); });
        },
    };
};
/**
 * KubernetesApi - object-oriented interface
 * @export
 * @class KubernetesApi
 * @extends {BaseAPI}
 */
var KubernetesApi = /** @class */ (function (_super) {
    __extends(KubernetesApi, _super);
    function KubernetesApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Deletes the K8s cluster specified  by its ID.
     * @summary Delete a Kubernetes Cluster by ID
     * @param {KubernetesApiK8sDeleteRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sDelete = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sDelete(requestParameters.k8sClusterId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves the K8s cluster specified by its ID.
     * @summary Get a Kubernetes Cluster by ID
     * @param {KubernetesApiK8sFindByClusterIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sFindByClusterId = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sFindByClusterId(requestParameters.k8sClusterId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves a list of all K8s clusters provisioned under your account.
     * @summary Get Kubernetes Clusters
     * @param {KubernetesApiK8sGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sGet = function (requestParameters, options) {
        var _this = this;
        if (requestParameters === void 0) { requestParameters = {}; }
        return exports.KubernetesApiFp(this.configuration).k8sGet(requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves the configuration file for the specified K8s cluster. You can define the format (YAML or JSON) of the returned file in the Accept header. By default, \'application/yaml\' is specified.
     * @summary Get Kubernetes Configuration File
     * @param {KubernetesApiK8sKubeconfigGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sKubeconfigGet = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sKubeconfigGet(requestParameters.k8sClusterId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Deletes the K8s node pool specified by its ID.
     * @summary Delete a Kubernetes Node Pool by ID
     * @param {KubernetesApiK8sNodepoolsDeleteRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsDelete = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsDelete(requestParameters.k8sClusterId, requestParameters.nodepoolId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves the K8s node pool specified by its ID.
     * @summary Get a Kubernetes Node Pool by ID
     * @param {KubernetesApiK8sNodepoolsFindByIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsFindById = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsFindById(requestParameters.k8sClusterId, requestParameters.nodepoolId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves a list of K8s node pools of a cluster specified by its ID.
     * @summary Get Kubernetes Node Pools
     * @param {KubernetesApiK8sNodepoolsGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsGet = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsGet(requestParameters.k8sClusterId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Deletes the K8s node specified by its ID.
     * @summary Delete a Kubernetes Node by ID
     * @param {KubernetesApiK8sNodepoolsNodesDeleteRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsNodesDelete = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsNodesDelete(requestParameters.k8sClusterId, requestParameters.nodepoolId, requestParameters.nodeId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves the K8s node specified by its ID.
     * @summary Get Kubernetes Node by ID
     * @param {KubernetesApiK8sNodepoolsNodesFindByIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsNodesFindById = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsNodesFindById(requestParameters.k8sClusterId, requestParameters.nodepoolId, requestParameters.nodeId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves the list of all K8s nodes of the specified node pool.
     * @summary Get Kubernetes Nodes
     * @param {KubernetesApiK8sNodepoolsNodesGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsNodesGet = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsNodesGet(requestParameters.k8sClusterId, requestParameters.nodepoolId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Recreates the K8s node specified by its ID.  If a node becomes unusable, Managed Kubernetes allows you to recreate it with a configuration based on the node pool template. Once the status is \'Active,\' all the pods from the failed node will be migrated to the new node. The node pool has an additional billable \'active\' node during this process.
     * @summary Recreate a Kubernetes Node by ID
     * @param {KubernetesApiK8sNodepoolsNodesReplacePostRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsNodesReplacePost = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsNodesReplacePost(requestParameters.k8sClusterId, requestParameters.nodepoolId, requestParameters.nodeId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Creates a node pool inside the specified K8s cluster.
     * @summary Create a Kubernetes Node Pool
     * @param {KubernetesApiK8sNodepoolsPostRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsPost = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsPost(requestParameters.k8sClusterId, requestParameters.kubernetesNodePool, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Modifies the K8s node pool specified by its ID.
     * @summary Modify a Kubernetes Node Pool by ID
     * @param {KubernetesApiK8sNodepoolsPutRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sNodepoolsPut = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sNodepoolsPut(requestParameters.k8sClusterId, requestParameters.nodepoolId, requestParameters.kubernetesNodePool, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Creates a K8s cluster provisioned under your account.
     * @summary Create a Kubernetes Cluster
     * @param {KubernetesApiK8sPostRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sPost = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sPost(requestParameters.kubernetesCluster, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Modifies the K8s cluster specified by its ID.
     * @summary Modify a Kubernetes Cluster by ID
     * @param {KubernetesApiK8sPutRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sPut = function (requestParameters, options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sPut(requestParameters.k8sClusterId, requestParameters.kubernetesCluster, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieves the current default K8s version to be used by the clusters and node pools.
     * @summary Get Default Kubernetes Version
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sVersionsDefaultGet = function (options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sVersionsDefaultGet(options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Lists available K8s versions.
     * @summary Get Kubernetes Versions
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof KubernetesApi
     */
    KubernetesApi.prototype.k8sVersionsGet = function (options) {
        var _this = this;
        return exports.KubernetesApiFp(this.configuration).k8sVersionsGet(options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    return KubernetesApi;
}(base_1.BaseAPI));
exports.KubernetesApi = KubernetesApi;
