// @ionos-cloud/sdk-nodejs/dist/api/lans-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LANsApi = exports.LANsApiFactory = exports.LANsApiFp = exports.LANsApiAxiosParamCreator = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
var base_1 = require("../base");
/**
 * LANsApi - axios parameter creator
 * @export
 */
exports.LANsApiAxiosParamCreator = function (configuration) {
    var _this = this;
    return {
        /**
         * Delete the specified LAN within the data center.
         * @summary Delete LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansDelete: function (datacenterId, lanId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansDelete.');
                            }
                            if (lanId === null || lanId === undefined) {
                                throw new base_1.RequiredError('lanId', 'Required parameter lanId was null or undefined when calling datacentersLansDelete.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans/{lanId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "lanId" + "}", encodeURIComponent(String(lanId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'DELETE' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieve the properties of the specified LAN within the data center.
         * @summary Retrieve LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansFindById: function (datacenterId, lanId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansFindById.');
                            }
                            if (lanId === null || lanId === undefined) {
                                throw new base_1.RequiredError('lanId', 'Required parameter lanId was null or undefined when calling datacentersLansFindById.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans/{lanId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "lanId" + "}", encodeURIComponent(String(lanId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * List all LANs within the data center.
         * @summary List LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansGet: function (datacenterId, pretty, depth, xContractNumber, offset, limit, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansGet.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((offset === undefined) && (configuration !== undefined)) {
                                offset = configuration.getDefaultParamValue('offset');
                            }
                            if (offset !== undefined) {
                                localVarQueryParameter['offset'] = offset;
                            }
                            if ((limit === undefined) && (configuration !== undefined)) {
                                limit = configuration.getDefaultParamValue('limit');
                            }
                            if (limit !== undefined) {
                                localVarQueryParameter['limit'] = limit;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieve the properties of the NIC, attached to the specified LAN.
         * @summary Retrieve attached NICs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {string} nicId The unique ID of the NIC.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsFindById: function (datacenterId, lanId, nicId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansNicsFindById.');
                            }
                            if (lanId === null || lanId === undefined) {
                                throw new base_1.RequiredError('lanId', 'Required parameter lanId was null or undefined when calling datacentersLansNicsFindById.');
                            }
                            if (nicId === null || nicId === undefined) {
                                throw new base_1.RequiredError('nicId', 'Required parameter nicId was null or undefined when calling datacentersLansNicsFindById.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans/{lanId}/nics/{nicId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "lanId" + "}", encodeURIComponent(String(lanId)))
                                .replace("{" + "nicId" + "}", encodeURIComponent(String(nicId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * List all NICs, attached to the specified LAN.
         * @summary List LAN members
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsGet: function (datacenterId, lanId, pretty, depth, xContractNumber, offset, limit, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansNicsGet.');
                            }
                            if (lanId === null || lanId === undefined) {
                                throw new base_1.RequiredError('lanId', 'Required parameter lanId was null or undefined when calling datacentersLansNicsGet.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans/{lanId}/nics"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "lanId" + "}", encodeURIComponent(String(lanId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((offset === undefined) && (configuration !== undefined)) {
                                offset = configuration.getDefaultParamValue('offset');
                            }
                            if (offset !== undefined) {
                                localVarQueryParameter['offset'] = offset;
                            }
                            if ((limit === undefined) && (configuration !== undefined)) {
                                limit = configuration.getDefaultParamValue('limit');
                            }
                            if (limit !== undefined) {
                                localVarQueryParameter['limit'] = limit;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Attach an existing NIC to the specified LAN.
         * @summary Attach NICs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {Nic} nic The NIC to be attached.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsPost: function (datacenterId, lanId, nic, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansNicsPost.');
                            }
                            if (lanId === null || lanId === undefined) {
                                throw new base_1.RequiredError('lanId', 'Required parameter lanId was null or undefined when calling datacentersLansNicsPost.');
                            }
                            if (nic === null || nic === undefined) {
                                throw new base_1.RequiredError('nic', 'Required parameter nic was null or undefined when calling datacentersLansNicsPost.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans/{lanId}/nics"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "lanId" + "}", encodeURIComponent(String(lanId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof nic !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(nic !== undefined ? nic : {})
                                : (nic || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Update the properties of the specified LAN within the data center.
         * @summary Partially modify LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {LanProperties} lan The properties of the LAN to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPatch: function (datacenterId, lanId, lan, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansPatch.');
                            }
                            if (lanId === null || lanId === undefined) {
                                throw new base_1.RequiredError('lanId', 'Required parameter lanId was null or undefined when calling datacentersLansPatch.');
                            }
                            if (lan === null || lan === undefined) {
                                throw new base_1.RequiredError('lan', 'Required parameter lan was null or undefined when calling datacentersLansPatch.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans/{lanId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "lanId" + "}", encodeURIComponent(String(lanId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'PATCH' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof lan !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(lan !== undefined ? lan : {})
                                : (lan || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Creates a LAN within the data center.
         * @summary Create LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {LanPost} lan The LAN to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPost: function (datacenterId, lan, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansPost.');
                            }
                            if (lan === null || lan === undefined) {
                                throw new base_1.RequiredError('lan', 'Required parameter lan was null or undefined when calling datacentersLansPost.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof lan !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(lan !== undefined ? lan : {})
                                : (lan || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Modify the properties of the specified LAN within the data center.
         * @summary Modify LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {Lan} lan The modified LAN
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPut: function (datacenterId, lanId, lan, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (datacenterId === null || datacenterId === undefined) {
                                throw new base_1.RequiredError('datacenterId', 'Required parameter datacenterId was null or undefined when calling datacentersLansPut.');
                            }
                            if (lanId === null || lanId === undefined) {
                                throw new base_1.RequiredError('lanId', 'Required parameter lanId was null or undefined when calling datacentersLansPut.');
                            }
                            if (lan === null || lan === undefined) {
                                throw new base_1.RequiredError('lan', 'Required parameter lan was null or undefined when calling datacentersLansPut.');
                            }
                            localVarPath = "/datacenters/{datacenterId}/lans/{lanId}"
                                .replace("{" + "datacenterId" + "}", encodeURIComponent(String(datacenterId)))
                                .replace("{" + "lanId" + "}", encodeURIComponent(String(lanId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'PUT' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof lan !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(lan !== undefined ? lan : {})
                                : (lan || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
    };
};
/**
 * LANsApi - functional programming interface
 * @export
 */
exports.LANsApiFp = function (configuration) {
    return {
        /**
         * Delete the specified LAN within the data center.
         * @summary Delete LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansDelete: function (datacenterId, lanId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansDelete(datacenterId, lanId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieve the properties of the specified LAN within the data center.
         * @summary Retrieve LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansFindById: function (datacenterId, lanId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansFindById(datacenterId, lanId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * List all LANs within the data center.
         * @summary List LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansGet: function (datacenterId, pretty, depth, xContractNumber, offset, limit, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansGet(datacenterId, pretty, depth, xContractNumber, offset, limit, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieve the properties of the NIC, attached to the specified LAN.
         * @summary Retrieve attached NICs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {string} nicId The unique ID of the NIC.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsFindById: function (datacenterId, lanId, nicId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansNicsFindById(datacenterId, lanId, nicId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * List all NICs, attached to the specified LAN.
         * @summary List LAN members
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsGet: function (datacenterId, lanId, pretty, depth, xContractNumber, offset, limit, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansNicsGet(datacenterId, lanId, pretty, depth, xContractNumber, offset, limit, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Attach an existing NIC to the specified LAN.
         * @summary Attach NICs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {Nic} nic The NIC to be attached.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsPost: function (datacenterId, lanId, nic, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansNicsPost(datacenterId, lanId, nic, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Update the properties of the specified LAN within the data center.
         * @summary Partially modify LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {LanProperties} lan The properties of the LAN to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPatch: function (datacenterId, lanId, lan, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansPatch(datacenterId, lanId, lan, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Creates a LAN within the data center.
         * @summary Create LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {LanPost} lan The LAN to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPost: function (datacenterId, lan, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansPost(datacenterId, lan, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Modify the properties of the specified LAN within the data center.
         * @summary Modify LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {Lan} lan The modified LAN
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPut: function (datacenterId, lanId, lan, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.LANsApiAxiosParamCreator(configuration).datacentersLansPut(datacenterId, lanId, lan, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
    };
};
/**
 * LANsApi - factory interface
 * @export
 */
exports.LANsApiFactory = function (configuration, basePath, axios) {
    return {
        /**
         * Delete the specified LAN within the data center.
         * @summary Delete LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansDelete: function (datacenterId, lanId, pretty, depth, xContractNumber, options) {
            return exports.LANsApiFp(configuration).datacentersLansDelete(datacenterId, lanId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieve the properties of the specified LAN within the data center.
         * @summary Retrieve LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansFindById: function (datacenterId, lanId, pretty, depth, xContractNumber, options) {
            return exports.LANsApiFp(configuration).datacentersLansFindById(datacenterId, lanId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * List all LANs within the data center.
         * @summary List LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansGet: function (datacenterId, pretty, depth, xContractNumber, offset, limit, options) {
            return exports.LANsApiFp(configuration).datacentersLansGet(datacenterId, pretty, depth, xContractNumber, offset, limit, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieve the properties of the NIC, attached to the specified LAN.
         * @summary Retrieve attached NICs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {string} nicId The unique ID of the NIC.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsFindById: function (datacenterId, lanId, nicId, pretty, depth, xContractNumber, options) {
            return exports.LANsApiFp(configuration).datacentersLansNicsFindById(datacenterId, lanId, nicId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * List all NICs, attached to the specified LAN.
         * @summary List LAN members
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {number} [offset] The first element (from the complete list of the elements) to include in the response (used together with &lt;b&gt;&lt;i&gt;limit&lt;/i&gt;&lt;/b&gt; for pagination).
         * @param {number} [limit] The maximum number of elements to return (use together with offset for pagination).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsGet: function (datacenterId, lanId, pretty, depth, xContractNumber, offset, limit, options) {
            return exports.LANsApiFp(configuration).datacentersLansNicsGet(datacenterId, lanId, pretty, depth, xContractNumber, offset, limit, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Attach an existing NIC to the specified LAN.
         * @summary Attach NICs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {Nic} nic The NIC to be attached.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansNicsPost: function (datacenterId, lanId, nic, pretty, depth, xContractNumber, options) {
            return exports.LANsApiFp(configuration).datacentersLansNicsPost(datacenterId, lanId, nic, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Update the properties of the specified LAN within the data center.
         * @summary Partially modify LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {LanProperties} lan The properties of the LAN to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPatch: function (datacenterId, lanId, lan, pretty, depth, xContractNumber, options) {
            return exports.LANsApiFp(configuration).datacentersLansPatch(datacenterId, lanId, lan, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Creates a LAN within the data center.
         * @summary Create LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {LanPost} lan The LAN to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPost: function (datacenterId, lan, pretty, depth, xContractNumber, options) {
            return exports.LANsApiFp(configuration).datacentersLansPost(datacenterId, lan, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Modify the properties of the specified LAN within the data center.
         * @summary Modify LANs
         * @param {string} datacenterId The unique ID of the data center.
         * @param {string} lanId The unique ID of the LAN.
         * @param {Lan} lan The modified LAN
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        datacentersLansPut: function (datacenterId, lanId, lan, pretty, depth, xContractNumber, options) {
            return exports.LANsApiFp(configuration).datacentersLansPut(datacenterId, lanId, lan, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
    };
};
/**
 * LANsApi - object-oriented interface
 * @export
 * @class LANsApi
 * @extends {BaseAPI}
 */
var LANsApi = /** @class */ (function (_super) {
    __extends(LANsApi, _super);
    function LANsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Delete the specified LAN within the data center.
     * @summary Delete LANs
     * @param {LANsApiDatacentersLansDeleteRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansDelete = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansDelete(requestParameters.datacenterId, requestParameters.lanId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieve the properties of the specified LAN within the data center.
     * @summary Retrieve LANs
     * @param {LANsApiDatacentersLansFindByIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansFindById = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansFindById(requestParameters.datacenterId, requestParameters.lanId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * List all LANs within the data center.
     * @summary List LANs
     * @param {LANsApiDatacentersLansGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansGet = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansGet(requestParameters.datacenterId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, requestParameters.offset, requestParameters.limit, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieve the properties of the NIC, attached to the specified LAN.
     * @summary Retrieve attached NICs
     * @param {LANsApiDatacentersLansNicsFindByIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansNicsFindById = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansNicsFindById(requestParameters.datacenterId, requestParameters.lanId, requestParameters.nicId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * List all NICs, attached to the specified LAN.
     * @summary List LAN members
     * @param {LANsApiDatacentersLansNicsGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansNicsGet = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansNicsGet(requestParameters.datacenterId, requestParameters.lanId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, requestParameters.offset, requestParameters.limit, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Attach an existing NIC to the specified LAN.
     * @summary Attach NICs
     * @param {LANsApiDatacentersLansNicsPostRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansNicsPost = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansNicsPost(requestParameters.datacenterId, requestParameters.lanId, requestParameters.nic, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Update the properties of the specified LAN within the data center.
     * @summary Partially modify LANs
     * @param {LANsApiDatacentersLansPatchRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansPatch = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansPatch(requestParameters.datacenterId, requestParameters.lanId, requestParameters.lan, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Creates a LAN within the data center.
     * @summary Create LANs
     * @param {LANsApiDatacentersLansPostRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansPost = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansPost(requestParameters.datacenterId, requestParameters.lan, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Modify the properties of the specified LAN within the data center.
     * @summary Modify LANs
     * @param {LANsApiDatacentersLansPutRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof LANsApi
     */
    LANsApi.prototype.datacentersLansPut = function (requestParameters, options) {
        var _this = this;
        return exports.LANsApiFp(this.configuration).datacentersLansPut(requestParameters.datacenterId, requestParameters.lanId, requestParameters.lan, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    return LANsApi;
}(base_1.BaseAPI));
exports.LANsApi = LANsApi;
