// @ionos-cloud/sdk-nodejs/dist/api/private-cross-connects-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateCrossConnectsApi = exports.PrivateCrossConnectsApiFactory = exports.PrivateCrossConnectsApiFp = exports.PrivateCrossConnectsApiAxiosParamCreator = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
var base_1 = require("../base");
/**
 * PrivateCrossConnectsApi - axios parameter creator
 * @export
 */
exports.PrivateCrossConnectsApiAxiosParamCreator = function (configuration) {
    var _this = this;
    return {
        /**
         * Remove the specified private Cross-Connect (only if not connected to any data centers).
         * @summary Delete private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsDelete: function (pccId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (pccId === null || pccId === undefined) {
                                throw new base_1.RequiredError('pccId', 'Required parameter pccId was null or undefined when calling pccsDelete.');
                            }
                            localVarPath = "/pccs/{pccId}"
                                .replace("{" + "pccId" + "}", encodeURIComponent(String(pccId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'DELETE' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Retrieve a private Cross-Connect by the resource ID. Cross-Connect ID is in the response body when the private Cross-Connect is created, and in the list of private Cross-Connects, returned by GET.
         * @summary Retrieve private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsFindById: function (pccId, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (pccId === null || pccId === undefined) {
                                throw new base_1.RequiredError('pccId', 'Required parameter pccId was null or undefined when calling pccsFindById.');
                            }
                            localVarPath = "/pccs/{pccId}"
                                .replace("{" + "pccId" + "}", encodeURIComponent(String(pccId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * List all private Cross-Connects for your account.
         * @summary List private Cross-Connects
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsGet: function (pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            localVarPath = "/pccs";
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'GET' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if (orderBy !== undefined) {
                                localVarQueryParameter['orderBy'] = orderBy;
                            }
                            if (maxResults !== undefined) {
                                localVarQueryParameter['maxResults'] = maxResults;
                            }
                            if (filters !== undefined) {
                                if (typeof (filters) === "object") {
                                    filters = new Map(Object.entries(filters));
                                }
                                filters.forEach(function (value, key) {
                                    localVarQueryParameter["filter." + key] = value;
                                });
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Update the properties of the specified private Cross-Connect.
         * @summary Partially modify private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {PrivateCrossConnectProperties} pcc The properties of the private Cross-Connect to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsPatch: function (pccId, pcc, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (pccId === null || pccId === undefined) {
                                throw new base_1.RequiredError('pccId', 'Required parameter pccId was null or undefined when calling pccsPatch.');
                            }
                            if (pcc === null || pcc === undefined) {
                                throw new base_1.RequiredError('pcc', 'Required parameter pcc was null or undefined when calling pccsPatch.');
                            }
                            localVarPath = "/pccs/{pccId}"
                                .replace("{" + "pccId" + "}", encodeURIComponent(String(pccId)));
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'PATCH' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof pcc !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(pcc !== undefined ? pcc : {})
                                : (pcc || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
        /**
         * Creates a private Cross-Connect.
         * @summary Create a Private Cross-Connect
         * @param {PrivateCrossConnect} pcc The private Cross-Connect to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsPost: function (pcc, pretty, depth, xContractNumber, options) {
            if (options === void 0) { options = {}; }
            return __awaiter(_this, void 0, void 0, function () {
                var localVarPath, localVarUrlObj, baseOptions, localVarRequestOptions, localVarHeaderParameter, localVarQueryParameter, localVarApiKeyValue, _a, queryParameters, key, key, headersFromBaseOptions, nonString, needsSerialization;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            if (pcc === null || pcc === undefined) {
                                throw new base_1.RequiredError('pcc', 'Required parameter pcc was null or undefined when calling pccsPost.');
                            }
                            localVarPath = "/pccs";
                            localVarUrlObj = new URL(localVarPath, 'https://example.com');
                            if (configuration) {
                                baseOptions = configuration.baseOptions;
                            }
                            localVarRequestOptions = __assign(__assign({ method: 'POST' }, baseOptions), options);
                            localVarHeaderParameter = {};
                            localVarQueryParameter = {};
                            // authentication Basic Authentication required
                            // http basic authentication required
                            if (configuration && (configuration.username || configuration.password)) {
                                localVarRequestOptions["auth"] = { username: configuration.username, password: configuration.password };
                            }
                            if (!(configuration && configuration.apiKey)) return [3 /*break*/, 5];
                            if (!(typeof configuration.apiKey === 'function')) return [3 /*break*/, 2];
                            return [4 /*yield*/, configuration.apiKey("Authorization")];
                        case 1:
                            _a = _b.sent();
                            return [3 /*break*/, 4];
                        case 2: return [4 /*yield*/, configuration.apiKey];
                        case 3:
                            _a = _b.sent();
                            _b.label = 4;
                        case 4:
                            localVarApiKeyValue = _a;
                            localVarHeaderParameter["Authorization"] = "Bearer " + localVarApiKeyValue;
                            _b.label = 5;
                        case 5:
                            if ((pretty === undefined) && (configuration !== undefined)) {
                                pretty = configuration.getDefaultParamValue('pretty');
                            }
                            if (pretty !== undefined) {
                                localVarQueryParameter['pretty'] = pretty;
                            }
                            if ((depth === undefined) && (configuration !== undefined)) {
                                depth = configuration.getDefaultParamValue('depth');
                            }
                            if (depth !== undefined) {
                                localVarQueryParameter['depth'] = depth;
                            }
                            if ((xContractNumber === undefined) && (configuration !== undefined)) {
                                xContractNumber = configuration.getDefaultParamValue('xContractNumber');
                            }
                            if (xContractNumber !== undefined && xContractNumber !== null) {
                                localVarHeaderParameter['X-Contract-Number'] = String(JSON.stringify(xContractNumber));
                            }
                            localVarHeaderParameter['Content-Type'] = 'application/json';
                            queryParameters = new URLSearchParams(localVarUrlObj.search);
                            for (key in localVarQueryParameter) {
                                queryParameters.set(key, localVarQueryParameter[key]);
                            }
                            for (key in options.query) {
                                queryParameters.set(key, options.query[key]);
                            }
                            localVarUrlObj.search = (new URLSearchParams(queryParameters)).toString();
                            headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                            localVarRequestOptions.headers = __assign(__assign(__assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                            nonString = typeof pcc !== 'string';
                            needsSerialization = nonString && configuration && configuration.isJsonMime
                                ? configuration.isJsonMime(localVarRequestOptions.headers['Content-Type'])
                                : nonString;
                            localVarRequestOptions.data = needsSerialization
                                ? JSON.stringify(pcc !== undefined ? pcc : {})
                                : (pcc || "");
                            return [2 /*return*/, {
                                    url: localVarUrlObj.pathname + localVarUrlObj.search + localVarUrlObj.hash,
                                    options: localVarRequestOptions,
                                }];
                    }
                });
            });
        },
    };
};
/**
 * PrivateCrossConnectsApi - functional programming interface
 * @export
 */
exports.PrivateCrossConnectsApiFp = function (configuration) {
    return {
        /**
         * Remove the specified private Cross-Connect (only if not connected to any data centers).
         * @summary Delete private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsDelete: function (pccId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.PrivateCrossConnectsApiAxiosParamCreator(configuration).pccsDelete(pccId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Retrieve a private Cross-Connect by the resource ID. Cross-Connect ID is in the response body when the private Cross-Connect is created, and in the list of private Cross-Connects, returned by GET.
         * @summary Retrieve private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsFindById: function (pccId, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.PrivateCrossConnectsApiAxiosParamCreator(configuration).pccsFindById(pccId, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * List all private Cross-Connects for your account.
         * @summary List private Cross-Connects
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {string} [orderBy] - Sorts the results alphanumerically in ascending order based on the specified property.
         * @param {number} [maxResults] - Limits the number of results returned.
         * @param {Map<string,string>} [filters] - Filters query parameters limit results to those containing a matching value for a specific property.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsGet: function (pretty, depth, xContractNumber, orderBy, maxResults, filters, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.PrivateCrossConnectsApiAxiosParamCreator(configuration).pccsGet(pretty, depth, xContractNumber, orderBy, maxResults, filters, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Update the properties of the specified private Cross-Connect.
         * @summary Partially modify private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {PrivateCrossConnectProperties} pcc The properties of the private Cross-Connect to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsPatch: function (pccId, pcc, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.PrivateCrossConnectsApiAxiosParamCreator(configuration).pccsPatch(pccId, pcc, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
        /**
         * Creates a private Cross-Connect.
         * @summary Create a Private Cross-Connect
         * @param {PrivateCrossConnect} pcc The private Cross-Connect to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsPost: function (pcc, pretty, depth, xContractNumber, options) {
            return __awaiter(this, void 0, void 0, function () {
                var axiosArgs;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, exports.PrivateCrossConnectsApiAxiosParamCreator(configuration).pccsPost(pcc, pretty, depth, xContractNumber, options)];
                        case 1:
                            axiosArgs = _a.sent();
                            return [2 /*return*/, base_1.runRequest(axiosArgs, configuration)];
                    }
                });
            });
        },
    };
};
/**
 * PrivateCrossConnectsApi - factory interface
 * @export
 */
exports.PrivateCrossConnectsApiFactory = function (configuration, basePath, axios) {
    return {
        /**
         * Remove the specified private Cross-Connect (only if not connected to any data centers).
         * @summary Delete private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsDelete: function (pccId, pretty, depth, xContractNumber, options) {
            return exports.PrivateCrossConnectsApiFp(configuration).pccsDelete(pccId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Retrieve a private Cross-Connect by the resource ID. Cross-Connect ID is in the response body when the private Cross-Connect is created, and in the list of private Cross-Connects, returned by GET.
         * @summary Retrieve private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsFindById: function (pccId, pretty, depth, xContractNumber, options) {
            return exports.PrivateCrossConnectsApiFp(configuration).pccsFindById(pccId, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * List all private Cross-Connects for your account.
         * @summary List private Cross-Connects
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsGet: function (pretty, depth, xContractNumber, options) {
            return exports.PrivateCrossConnectsApiFp(configuration).pccsGet(pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Update the properties of the specified private Cross-Connect.
         * @summary Partially modify private Cross-Connects
         * @param {string} pccId The unique ID of the private Cross-Connect.
         * @param {PrivateCrossConnectProperties} pcc The properties of the private Cross-Connect to be updated.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsPatch: function (pccId, pcc, pretty, depth, xContractNumber, options) {
            return exports.PrivateCrossConnectsApiFp(configuration).pccsPatch(pccId, pcc, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
        /**
         * Creates a private Cross-Connect.
         * @summary Create a Private Cross-Connect
         * @param {PrivateCrossConnect} pcc The private Cross-Connect to create.
         * @param {boolean} [pretty] Controls whether the response is pretty-printed (with indentations and new lines).
         * @param {number} [depth] Controls the detail depth of the response objects.  GET /datacenters/[ID]  - depth&#x3D;0: Only direct properties are included; children (servers and other elements) are not included.  - depth&#x3D;1: Direct properties and children references are included.  - depth&#x3D;2: Direct properties and children properties are included.  - depth&#x3D;3: Direct properties and children properties and children\&#39;s children are included.  - depth&#x3D;... and so on
         * @param {number} [xContractNumber] Users with multiple contracts must provide the contract number, for which all API requests are to be executed.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pccsPost: function (pcc, pretty, depth, xContractNumber, options) {
            return exports.PrivateCrossConnectsApiFp(configuration).pccsPost(pcc, pretty, depth, xContractNumber, options).then(function (request) { return request(axios, basePath); });
        },
    };
};
/**
 * PrivateCrossConnectsApi - object-oriented interface
 * @export
 * @class PrivateCrossConnectsApi
 * @extends {BaseAPI}
 */
var PrivateCrossConnectsApi = /** @class */ (function (_super) {
    __extends(PrivateCrossConnectsApi, _super);
    function PrivateCrossConnectsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Remove the specified private Cross-Connect (only if not connected to any data centers).
     * @summary Delete private Cross-Connects
     * @param {PrivateCrossConnectsApiPccsDeleteRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PrivateCrossConnectsApi
     */
    PrivateCrossConnectsApi.prototype.pccsDelete = function (requestParameters, options) {
        var _this = this;
        return exports.PrivateCrossConnectsApiFp(this.configuration).pccsDelete(requestParameters.pccId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Retrieve a private Cross-Connect by the resource ID. Cross-Connect ID is in the response body when the private Cross-Connect is created, and in the list of private Cross-Connects, returned by GET.
     * @summary Retrieve private Cross-Connects
     * @param {PrivateCrossConnectsApiPccsFindByIdRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PrivateCrossConnectsApi
     */
    PrivateCrossConnectsApi.prototype.pccsFindById = function (requestParameters, options) {
        var _this = this;
        return exports.PrivateCrossConnectsApiFp(this.configuration).pccsFindById(requestParameters.pccId, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * List all private Cross-Connects for your account.
     * @summary List private Cross-Connects
     * @param {PrivateCrossConnectsApiPccsGetRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PrivateCrossConnectsApi
     */
    PrivateCrossConnectsApi.prototype.pccsGet = function (requestParameters, options) {
        var _this = this;
        if (requestParameters === void 0) { requestParameters = {}; }
        return exports.PrivateCrossConnectsApiFp(this.configuration).pccsGet(requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, requestParameters.orderBy, requestParameters.maxResults, requestParameters.filters, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Update the properties of the specified private Cross-Connect.
     * @summary Partially modify private Cross-Connects
     * @param {PrivateCrossConnectsApiPccsPatchRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PrivateCrossConnectsApi
     */
    PrivateCrossConnectsApi.prototype.pccsPatch = function (requestParameters, options) {
        var _this = this;
        return exports.PrivateCrossConnectsApiFp(this.configuration).pccsPatch(requestParameters.pccId, requestParameters.pcc, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    /**
     * Creates a private Cross-Connect.
     * @summary Create a Private Cross-Connect
     * @param {PrivateCrossConnectsApiPccsPostRequest} requestParameters Request parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PrivateCrossConnectsApi
     */
    PrivateCrossConnectsApi.prototype.pccsPost = function (requestParameters, options) {
        var _this = this;
        return exports.PrivateCrossConnectsApiFp(this.configuration).pccsPost(requestParameters.pcc, requestParameters.pretty, requestParameters.depth, requestParameters.xContractNumber, options).then(function (request) { return request(_this.axios, _this.basePath); });
    };
    return PrivateCrossConnectsApi;
}(base_1.BaseAPI));
exports.PrivateCrossConnectsApi = PrivateCrossConnectsApi;
