// @ionos-cloud/sdk-nodejs/dist/model/datacenter-element-metadata.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatacenterElementMetadataStateEnum = void 0;
/**
    * @export
    * @enum {string}
    */
var DatacenterElementMetadataStateEnum;
(function (DatacenterElementMetadataStateEnum) {
    DatacenterElementMetadataStateEnum["Available"] = "AVAILABLE";
    DatacenterElementMetadataStateEnum["Inactive"] = "INACTIVE";
    DatacenterElementMetadataStateEnum["Busy"] = "BUSY";
    DatacenterElementMetadataStateEnum["Deploying"] = "DEPLOYING";
    DatacenterElementMetadataStateEnum["Active"] = "ACTIVE";
    DatacenterElementMetadataStateEnum["Failed"] = "FAILED";
    DatacenterElementMetadataStateEnum["Suspended"] = "SUSPENDED";
    DatacenterElementMetadataStateEnum["FailedSuspended"] = "FAILED_SUSPENDED";
    DatacenterElementMetadataStateEnum["Updating"] = "UPDATING";
    DatacenterElementMetadataStateEnum["FailedUpdating"] = "FAILED_UPDATING";
    DatacenterElementMetadataStateEnum["Destroying"] = "DESTROYING";
    DatacenterElementMetadataStateEnum["FailedDestroying"] = "FAILED_DESTROYING";
    DatacenterElementMetadataStateEnum["Terminated"] = "TERMINATED";
    DatacenterElementMetadataStateEnum["Hibernating"] = "HIBERNATING";
    DatacenterElementMetadataStateEnum["FailedHibernating"] = "FAILED_HIBERNATING";
    DatacenterElementMetadataStateEnum["Maintenance"] = "MAINTENANCE";
    DatacenterElementMetadataStateEnum["FailedMaintenance"] = "FAILED_MAINTENANCE";
    DatacenterElementMetadataStateEnum["Unknown"] = "UNKNOWN";
})(DatacenterElementMetadataStateEnum = exports.DatacenterElementMetadataStateEnum || (exports.DatacenterElementMetadataStateEnum = {}));
