// @ionos-cloud/sdk-nodejs/dist/model/firewallrule-properties.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirewallrulePropertiesTypeEnum = exports.FirewallrulePropertiesProtocolEnum = exports.FirewallrulePropertiesIpVersionEnum = void 0;
/**
    * @export
    * @enum {string}
    */
var FirewallrulePropertiesIpVersionEnum;
(function (FirewallrulePropertiesIpVersionEnum) {
    FirewallrulePropertiesIpVersionEnum["Ipv4"] = "IPv4";
    FirewallrulePropertiesIpVersionEnum["Ipv6"] = "IPv6";
})(FirewallrulePropertiesIpVersionEnum = exports.FirewallrulePropertiesIpVersionEnum || (exports.FirewallrulePropertiesIpVersionEnum = {}));
/**
    * @export
    * @enum {string}
    */
var FirewallrulePropertiesProtocolEnum;
(function (FirewallrulePropertiesProtocolEnum) {
    FirewallrulePropertiesProtocolEnum["Tcp"] = "TCP";
    FirewallrulePropertiesProtocolEnum["Udp"] = "UDP";
    FirewallrulePropertiesProtocolEnum["Icmp"] = "ICMP";
    FirewallrulePropertiesProtocolEnum["Icmpv6"] = "ICMPv6";
    FirewallrulePropertiesProtocolEnum["Any"] = "ANY";
})(FirewallrulePropertiesProtocolEnum = exports.FirewallrulePropertiesProtocolEnum || (exports.FirewallrulePropertiesProtocolEnum = {}));
/**
    * @export
    * @enum {string}
    */
var FirewallrulePropertiesTypeEnum;
(function (FirewallrulePropertiesTypeEnum) {
    FirewallrulePropertiesTypeEnum["Ingress"] = "INGRESS";
    FirewallrulePropertiesTypeEnum["Egress"] = "EGRESS";
})(FirewallrulePropertiesTypeEnum = exports.FirewallrulePropertiesTypeEnum || (exports.FirewallrulePropertiesTypeEnum = {}));
