// @ionos-cloud/sdk-nodejs/dist/model/image-properties.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImagePropertiesLicenceTypeEnum = exports.ImagePropertiesImageTypeEnum = exports.ImagePropertiesCloudInitEnum = void 0;
/**
    * @export
    * @enum {string}
    */
var ImagePropertiesCloudInitEnum;
(function (ImagePropertiesCloudInitEnum) {
    ImagePropertiesCloudInitEnum["None"] = "NONE";
    ImagePropertiesCloudInitEnum["V1"] = "V1";
})(ImagePropertiesCloudInitEnum = exports.ImagePropertiesCloudInitEnum || (exports.ImagePropertiesCloudInitEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ImagePropertiesImageTypeEnum;
(function (ImagePropertiesImageTypeEnum) {
    ImagePropertiesImageTypeEnum["Hdd"] = "HDD";
    ImagePropertiesImageTypeEnum["Cdrom"] = "CDROM";
    ImagePropertiesImageTypeEnum["Unknown"] = "UNKNOWN";
})(ImagePropertiesImageTypeEnum = exports.ImagePropertiesImageTypeEnum || (exports.ImagePropertiesImageTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ImagePropertiesLicenceTypeEnum;
(function (ImagePropertiesLicenceTypeEnum) {
    ImagePropertiesLicenceTypeEnum["Unknown"] = "UNKNOWN";
    ImagePropertiesLicenceTypeEnum["Windows"] = "WINDOWS";
    ImagePropertiesLicenceTypeEnum["Windows2016"] = "WINDOWS2016";
    ImagePropertiesLicenceTypeEnum["Windows2022"] = "WINDOWS2022";
    ImagePropertiesLicenceTypeEnum["Rhel"] = "RHEL";
    ImagePropertiesLicenceTypeEnum["Linux"] = "LINUX";
    ImagePropertiesLicenceTypeEnum["Other"] = "OTHER";
})(ImagePropertiesLicenceTypeEnum = exports.ImagePropertiesLicenceTypeEnum || (exports.ImagePropertiesLicenceTypeEnum = {}));
