// @ionos-cloud/sdk-nodejs/dist/model/server-properties.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerPropertiesVmStateEnum = exports.ServerPropertiesAvailabilityZoneEnum = void 0;
/**
    * @export
    * @enum {string}
    */
var ServerPropertiesAvailabilityZoneEnum;
(function (ServerPropertiesAvailabilityZoneEnum) {
    ServerPropertiesAvailabilityZoneEnum["Auto"] = "AUTO";
    ServerPropertiesAvailabilityZoneEnum["Zone1"] = "ZONE_1";
    ServerPropertiesAvailabilityZoneEnum["Zone2"] = "ZONE_2";
})(ServerPropertiesAvailabilityZoneEnum = exports.ServerPropertiesAvailabilityZoneEnum || (exports.ServerPropertiesAvailabilityZoneEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ServerPropertiesVmStateEnum;
(function (ServerPropertiesVmStateEnum) {
    ServerPropertiesVmStateEnum["Nostate"] = "NOSTATE";
    ServerPropertiesVmStateEnum["Running"] = "RUNNING";
    ServerPropertiesVmStateEnum["Blocked"] = "BLOCKED";
    ServerPropertiesVmStateEnum["Paused"] = "PAUSED";
    ServerPropertiesVmStateEnum["Shutdown"] = "SHUTDOWN";
    ServerPropertiesVmStateEnum["Shutoff"] = "SHUTOFF";
    ServerPropertiesVmStateEnum["Crashed"] = "CRASHED";
    ServerPropertiesVmStateEnum["Suspended"] = "SUSPENDED";
})(ServerPropertiesVmStateEnum = exports.ServerPropertiesVmStateEnum || (exports.ServerPropertiesVmStateEnum = {}));
