// @ionos-cloud/sdk-nodejs/dist/model/snapshot-properties.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SnapshotPropertiesLicenceTypeEnum = void 0;
/**
    * @export
    * @enum {string}
    */
var SnapshotPropertiesLicenceTypeEnum;
(function (SnapshotPropertiesLicenceTypeEnum) {
    SnapshotPropertiesLicenceTypeEnum["Unknown"] = "UNKNOWN";
    SnapshotPropertiesLicenceTypeEnum["Windows"] = "WINDOWS";
    SnapshotPropertiesLicenceTypeEnum["Windows2016"] = "WINDOWS2016";
    SnapshotPropertiesLicenceTypeEnum["Windows2022"] = "WINDOWS2022";
    SnapshotPropertiesLicenceTypeEnum["Rhel"] = "RHEL";
    SnapshotPropertiesLicenceTypeEnum["Linux"] = "LINUX";
    SnapshotPropertiesLicenceTypeEnum["Other"] = "OTHER";
})(SnapshotPropertiesLicenceTypeEnum = exports.SnapshotPropertiesLicenceTypeEnum || (exports.SnapshotPropertiesLicenceTypeEnum = {}));
