// @ionos-cloud/sdk-nodejs/dist/model/type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = void 0;
/**
 *
 * @export
 * @enum {string}
 */
var Type;
(function (Type) {
    Type["Datacenter"] = "datacenter";
    Type["Server"] = "server";
    Type["Volume"] = "volume";
    Type["Nic"] = "nic";
    Type["Loadbalancer"] = "loadbalancer";
    Type["Location"] = "location";
    Type["FirewallRule"] = "firewall-rule";
    Type["FlowLog"] = "flow-log";
    Type["Image"] = "image";
    Type["Snapshot"] = "snapshot";
    Type["Lan"] = "lan";
    Type["Ipblock"] = "ipblock";
    Type["Pcc"] = "pcc";
    Type["Contract"] = "contract";
    Type["User"] = "user";
    Type["Group"] = "group";
    Type["Collection"] = "collection";
    Type["Resource"] = "resource";
    Type["Request"] = "request";
    Type["RequestStatus"] = "request-status";
    Type["S3key"] = "s3key";
    Type["Backupunit"] = "backupunit";
    Type["Label"] = "label";
    Type["K8s"] = "k8s";
    Type["Nodepool"] = "nodepool";
    Type["Template"] = "template";
    Type["Networkloadbalancer"] = "networkloadbalancer";
    Type["ForwardingRule"] = "forwarding-rule";
    Type["Natgateway"] = "natgateway";
    Type["NatgatewayRule"] = "natgateway-rule";
    Type["Node"] = "node";
    Type["Applicationloadbalancer"] = "applicationloadbalancer";
    Type["TargetGroup"] = "target-group";
})(Type = exports.Type || (exports.Type = {}));
