// @ionos-cloud/sdk-nodejs/dist/model/volume-properties.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * CLOUD API
 * IONOS Enterprise-grade Infrastructure as a Service (IaaS) solutions can be managed through the Cloud API, in addition or as an alternative to the \"Data Center Designer\" (DCD) browser-based tool.    Both methods employ consistent concepts and features, deliver similar power and flexibility, and can be used to perform a multitude of management tasks, including adding servers, volumes, configuring networks, and so on.
 *
 * The version of the OpenAPI document: 6.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumePropertiesTypeEnum = exports.VolumePropertiesLicenceTypeEnum = exports.VolumePropertiesBusEnum = exports.VolumePropertiesBootOrderEnum = exports.VolumePropertiesAvailabilityZoneEnum = void 0;
/**
    * @export
    * @enum {string}
    */
var VolumePropertiesAvailabilityZoneEnum;
(function (VolumePropertiesAvailabilityZoneEnum) {
    VolumePropertiesAvailabilityZoneEnum["Auto"] = "AUTO";
    VolumePropertiesAvailabilityZoneEnum["Zone1"] = "ZONE_1";
    VolumePropertiesAvailabilityZoneEnum["Zone2"] = "ZONE_2";
    VolumePropertiesAvailabilityZoneEnum["Zone3"] = "ZONE_3";
})(VolumePropertiesAvailabilityZoneEnum = exports.VolumePropertiesAvailabilityZoneEnum || (exports.VolumePropertiesAvailabilityZoneEnum = {}));
/**
    * @export
    * @enum {string}
    */
var VolumePropertiesBootOrderEnum;
(function (VolumePropertiesBootOrderEnum) {
    VolumePropertiesBootOrderEnum["Auto"] = "AUTO";
    VolumePropertiesBootOrderEnum["None"] = "NONE";
    VolumePropertiesBootOrderEnum["Primary"] = "PRIMARY";
})(VolumePropertiesBootOrderEnum = exports.VolumePropertiesBootOrderEnum || (exports.VolumePropertiesBootOrderEnum = {}));
/**
    * @export
    * @enum {string}
    */
var VolumePropertiesBusEnum;
(function (VolumePropertiesBusEnum) {
    VolumePropertiesBusEnum["Virtio"] = "VIRTIO";
    VolumePropertiesBusEnum["Ide"] = "IDE";
    VolumePropertiesBusEnum["Unknown"] = "UNKNOWN";
})(VolumePropertiesBusEnum = exports.VolumePropertiesBusEnum || (exports.VolumePropertiesBusEnum = {}));
/**
    * @export
    * @enum {string}
    */
var VolumePropertiesLicenceTypeEnum;
(function (VolumePropertiesLicenceTypeEnum) {
    VolumePropertiesLicenceTypeEnum["Unknown"] = "UNKNOWN";
    VolumePropertiesLicenceTypeEnum["Windows"] = "WINDOWS";
    VolumePropertiesLicenceTypeEnum["Windows2016"] = "WINDOWS2016";
    VolumePropertiesLicenceTypeEnum["Windows2022"] = "WINDOWS2022";
    VolumePropertiesLicenceTypeEnum["Rhel"] = "RHEL";
    VolumePropertiesLicenceTypeEnum["Linux"] = "LINUX";
    VolumePropertiesLicenceTypeEnum["Other"] = "OTHER";
})(VolumePropertiesLicenceTypeEnum = exports.VolumePropertiesLicenceTypeEnum || (exports.VolumePropertiesLicenceTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var VolumePropertiesTypeEnum;
(function (VolumePropertiesTypeEnum) {
    VolumePropertiesTypeEnum["Hdd"] = "HDD";
    VolumePropertiesTypeEnum["Ssd"] = "SSD";
    VolumePropertiesTypeEnum["SsdStandard"] = "SSD Standard";
    VolumePropertiesTypeEnum["SsdPremium"] = "SSD Premium";
    VolumePropertiesTypeEnum["Das"] = "DAS";
    VolumePropertiesTypeEnum["Iso"] = "ISO";
})(VolumePropertiesTypeEnum = exports.VolumePropertiesTypeEnum || (exports.VolumePropertiesTypeEnum = {}));
